### PR TITLE
Fix for missing kotlin project coverage reports.

### PR DESCRIPTION
**Pull request by yifanyang**

### CODE REVIEW

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy [hunk 1]
```diff
@@ -54,22 +54,25 @@ public class GenerateMeasurementsTask extends DefaultTask {
...
         try {
             def report = parser.parse(path)
-            def name = report.@name
             def lineCoverage = report.counter.find { it.@type == 'LINE' }
             if (lineCoverage) {
                 def covered = Double.parseDouble(lineCoverage.@covered.text())
                 def missed = Double.parseDouble(lineCoverage.@missed.text())
                 def percent = covered / (covered + missed)
-                return new Tuple(name, percent)
+                return percent
```
Comment (Corrob) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: How about just

return covered / (covered + missed)

?

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy [hunk 2]
```diff
@@ -84,12 +87,23 @@ public class GenerateMeasurementsTask extends DefaultTask {
...
+
+                // 'firebase-common-ktx' and 'firebase-firestore-ktx' has been
+                // made a subproject of their corresponding java projects in PR
+                // #409 and #426. Old mappings are kept for backward
+                // compatibility reason.
+                'firebase-common:ktx': 10,
+                'firebase-firestore:ktx': 11,
+
+                // 'firebase-storage:test-app' was added in PR #303.
+                'firebase-storage:test-app': 12
```
Comment (allisonbm92) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: This is a test app that is only used for instrumentation tests. I don't think we need to measure coverage for it.

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy [hunk 3]
```diff
@@ -54,22 +54,25 @@ public class GenerateMeasurementsTask extends DefaultTask {
...
                 def percent = covered / (covered + missed)
-                return new Tuple(name, percent)
+                return percent
             } else {
                 throw new IllegalStateException("Cannot find line coverage section in the report: $path.")
             }
         } catch (FileNotFoundException e) {
             project.logger.warn("Cannot find coverage report for project: $_project.")
-            return new Tuple(_project.name, 0)
+            return 0
```
Comment (allisonbm92) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: Should we treat this as an error?

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy (hunk 1 above)
Comment (yifanyang) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: Done.

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy (hunk 2 above)
Comment (yifanyang) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: Done. Removed.

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy (hunk 3 above)
Comment (yifanyang) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: Maybe.

I guess:
* if it is a problem with the coverage code itself, then it makes sense to throw an exception here;
* if it is due to an incorrectly configured project, then the code should just omit it and run through.

I haven't met a case in real life where this code path got hit, so as of now I don't know.

Comment (allisonbm92) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: What happens if the tests fail? Is there a coverage report? Or is that skipped due to the tests terminating early?

Comment (yifanyang) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: The report generation task depends on the unit test task. Should the unit test fail, gradle will stop early and won't get to the coverage report task.

Comment (allisonbm92) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy]: Right, that makes sense.

**Pull request merged**